File: 백준/Silver/28278. 스택 2/스택 2.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n")
// let input = fs.readFileSync("./EX.txt").toString().trim().split('\n')
input = input.map(x => {const line = x.split(" "); return [parseInt(line[0]), parseInt(line[1])]})

let stack = []
let result = []
input.slice(1, ).forEach(x => {
    let n = x[0]

    if (n === 1) {
        stack.push(x[1])
    } else if (n === 2) {
        if (stack.length) {
            x = stack.pop()
            result.push(x)
        } else {
            result.push(-1)
        }
    } else if (n === 3) {
        result.push(stack.length)
    } else if (n === 4) {
        if (stack.length) {
            result.push(0)
        } else {
            result.push(1)
        }
    } else if (n === 5) {
        if (stack.length) {
            result.push(stack[stack.length-1])
        } else {
            result.push(-1)
        }
    }

})
console.log(result.join("\n"))

File: 백준/Silver/28278. 스택 2/스택 2.py
import sys, math
input = sys.stdin.readline

N = int(input())

stack = []
for _ in range(N):
    arr = [int(x) for x in input().split()]
    num = arr[0]

    if num == 1:
        stack.append(arr[1])
    elif num == 2:
        if stack:
            x = stack.pop()
            print(x)
        else:
            print(-1)
    elif num == 3:
        print(len(stack))
    elif num == 4:
        if stack:
            print(0)
        else:
            print(1)
    elif num == 5:
        if stack:
            print(stack[-1])
        else:
            print(-1)